README.md

```diff
@@ -53,7 +53,42 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre.
+
+#### the complete code base
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code is PEP8 conform.
+
+#### lib Tests (`onyo/lib/`)
+- they follow Onyos code conventions, see `"the complete code base"
+- The tests proof stability of the repository class and the rest of the code to
+  offer a stable base for the consumers of their code (i.e. CLI, demos)
+- all public functions of all classes are tested:
```

Comment:
Public properties should also be tested. I can't think of a good word that covers all. Perhaps "Public endpoints"?

Reply:
Nah, I feel like trying to kill these to birds with one stone does not really bring the enhancement in clearity we are looking for. I think I will just add the words "public properties"!